packages/random/random.js

```diff
@@ -35,214 +39,205 @@ var Alea = function () {
+const BASE64_CHARS = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ' +
+  '0123456789-_';
 
 // `type` is one of `RandomGenerator.Type` as defined below.
 //
 // options:
 // - seeds: (required, only for RandomGenerator.Type.ALEA) an array
 //   whose items will be `toString`ed and used as the seed to the Alea
 //   algorithm
-var RandomGenerator = function (type, options) {
-  var self = this;
-  self.type = type;
+class RandomGenerator {
+  constructor (type, { seeds = [] } = {}) {
+    this.type = type;
+
+    if (!RandomGenerator.Type[type]) {
+      throw new Error(`Unknown random generator type: ${type}`);
+    }
 
-  if (!RandomGenerator.Type[type]) {
-    throw new Error("Unknown random generator type: " + type);
+    if (type === RandomGenerator.Type.ALEA) {
+      if (!seeds) {
+        throw new Error('No seeds were provided for Alea PRNG');
+      }
+      this.alea = Alea(seeds);
+    }
   }
 
-  if (type === RandomGenerator.Type.ALEA) {
-    if (!options.seeds) {
-      throw new Error("No seeds were provided for Alea PRNG");
+  /**
+   * @name Random.fraction
+   * @summary Return a number between 0 and 1, like `Math.random`.
+   * @locus Anywhere
+   */
+  fraction () {
+    switch (this.type) {
+    case RandomGenerator.Type.ALEA:
+      return this.alea();
+    case RandomGenerator.Type.NODE_CRYPTO: {
+      const numerator = Number.parseInt(this.hexString(8), 16);
+      return numerator * 2.3283064365386963e-10; // 2^-32
+    }
+    case RandomGenerator.Type.BROWSER_CRYPTO: {
+      const array = new Uint32Array(1);
+      window.crypto.getRandomValues(array);
+      return array[0] * 2.3283064365386963e-10; // 2^-32
+    }
+    default:
+      throw new Error(`Unknown random generator type: ${this.type}`);
     }
-    self.alea = Alea.apply(null, options.seeds);
   }
-};
+
+  /**
+   * @name Random.hexString
+   * @summary Return a random string of `n` hexadecimal digits.
+   * @locus Anywhere
+   * @param {Number} n Length of the string
+   */
+  hexString (digits) {
+    if (this.type === RandomGenerator.Type.NODE_CRYPTO) {
+      const numBytes = Math.ceil(digits / 2);
+      let bytes;
+      // Try to get cryptographically strong randomness. Fall back to
+      // non-cryptographically strong if not available.
+      try {
+        bytes = nodeCrypto.randomBytes(numBytes);
+      } catch (e) {
+        // XXX should re-throw any error except insufficient entropy
+        bytes = nodeCrypto.pseudoRandomBytes(numBytes);
+      }
+      const result = bytes.toString('hex');
+      // If the number of digits is odd, we'll have generated an extra 4 bits
+      // of randomness, so we need to trim the last digit.
+      return result.substring(0, digits);
+    } else {
+      return this._randomString(digits, '0123456789abcdef');
+    }
+  }
+
+  _randomString (charsCount, alphabet) {
+    return [...Array(charsCount)].map(this.choice.bind(this, alphabet)).join('');
```

Comment:
Ok, although this would only get called for Mongo ids on the client side, I agree it might be case of wanting to use the new stuff without much reason. I'll change it to what you suggest.